### DISCUSSION TITLE
Trying to understand the way to use spancat on top of pretrained NER model(en core web sm).

### DISCUSSION THREAD

**Discussion by GJ1700** [explosion/spaCy#8989] [Help: Model Advice] [answered]
I'm trying to understand the way of enabling spancat on top of NER labels in prodigy.

About the problem: We are trying to solve the Entity Level Sentiment problem. So for e.g. we have text as "I like Joseph but I dont like Sam". So here Joseph and Sam will be identified as person entity by NER, on top this person entities
 I want to use span categorizer to categorize the sentiment as "POSITIVE" for JOSEPH and "NEGATIVE" for SAM.

The problem I'm facing now is, I am not able to generate NER entities in prodigy. Is there a way/command to do it in prodigy?

I have attached the command I am using for prodigy spanCat.
<img width="904" alt="Capture" src="https://user-images.githubusercontent.com/59384434/130043928-e2a4a995-5017-4cd3-b903-9c253b1aa227.PNG">

**Comment by svlandeg** [comment, marked as answer]
Hi!

I understand the context and what you're trying to do, but I'm not sure I understand the problem you're facing. Could you elaborate on "I am not able to generate NER entities"? 

If you want to annotate the sentiment spans while seeing the annotated NER entities, what you could do is run the NER model over the input texts and make sure they are included in the input jsonl, something like:
```
{"text": Uber's Lesson: Silicon Valley's Start-Up Machine Needs Fixing, .
"tokens":[{"text":"Uber","start":0,"end":4,"id":0},...,{"text":"Silicon","start":15,"end":22,"id":4},{"text":"Valley","start":23,"end":29,"id":5},...],
"spans":[{"start":0,"end":4,"token_start":0,"token_end":0,"label":"ORG"},{"start":15,"end":29,"token_start":4,"token_end":5,"label":"LOCATION"}],
...}
```
Then when you use this JSONL file as input for `spans.manual`, the NER entities will be highlighted.

**Reply by GJ1700** [reply]
Hey @svlandeg, Thanks for taking time and provide the best possible answer. 

I think you got it right, I have attached the screenshot below:

<img width="598" alt="Capture" src="https://user-images.githubusercontent.com/59384434/130266131-4e0c7814-5470-43fa-8c6a-e18a2eecb703.PNG">

So here I want the **SAP** to be highlighted as **"ORG"** by using spacy's **"en_core_web_lg"** and then I would be assigining "POSITIVE" span category to it. 

I would definitely try the solution you mentioned above. This is just for clarification.

Thanks a ton, Cheers!!

**Comment by lnatspacy** [comment]
Alternatively (this is how I did it) you can convert the entities in your training data into spans and then train a spancat model, that will predict pretty much the same as your ner model, only it'll predict spans instead of entities. 
Once you have that, you can use spans.correct (available in the latest version) which gives you the exact experience you were looking for :)

**Reply by svlandeg** [reply]
That's a good idea as well, thanks for sharing!

**Reply by GJ1700** [reply]
Thanks @lnatspacy & @svlandeg. I will definitely try both the approaches and let u guys know.
Cheers!!!

**Reply by lnatspacy** [reply]
Check this out https://support.prodi.gy/t/preannotated-spans-in-input-json-not-showing-up/4601/7 
I wrote a custom recipe that let's you use a ner model to annotate spancat data. On top of that you can also use patterns and a model at the same time to help you annotate your data.
